src/rum/instrumentation/DdRumReactNavigationTracking.tsx

```diff
@@ -6,29 +6,61 @@
 
 import type { EventArg, NavigationContainerRef, Route } from "@react-navigation/native";
 import { DdRum } from '../../index';
+import { AppState, AppStateStatus } from 'react-native';
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 declare type NavigationListener = (event: EventArg<string, boolean, any>) => void | null
 
+declare type AppStateListener = (appStateStatus: AppStateStatus) => void | null
+
 /**
  * Provides RUM integration for the [ReactNavigation](https://reactnavigation.org/) API.
  */
 export default class DdRumReactNavigationTracking {
-    private static registeredContainers: WeakSet<NavigationContainerRef> = new WeakSet();
+
+    private static registeredContainer: NavigationContainerRef | null;
 
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     private static navigationStateChangeListener: NavigationListener;
 
+    private static appStateListener: AppStateListener;
+
     /**
      * Starts tracking the NavigationContainer and sends a RUM View event every time the navigation route changed.
      * @param navigationRef the reference to the real NavigationContainer.
      */
     static startTrackingViews(navigationRef: NavigationContainerRef | null): void {
-        if (navigationRef != null && !this.registeredContainers.has(navigationRef)) {
+        if (navigationRef == null) {
+            return;
+        }
+
+        if (this.registeredContainer != null && this.registeredContainer !== navigationRef) {
+            console.error('Cannot track new navigation container while another one is still tracked');
```

Comment:
not sure if we should notify user here, or just silently override container tracked.